File: uLab/Assets/Scripts/Test/DialogWindow.cs


using Locke;
using Locke.ui;
using UnityEngine;
using UnityEngine.UI;

public class DialogWindow : IWindow
{
	protected override void OnEnter(IContext context)
	{
		Button button = this.FindWidget<Button>("buttons/button1");
		UIEventListener.Get(button.gameObject).onClick = onOkClick;

		Button back = FindWidget<Button>("buttons/back");
		back.onClick.AddListener(OnBackClick);
	}

	protected override void OnExit(IContext context)
	{
	}

	protected override void OnPause(IContext context)
	{
	}

	protected override void OnResume(IContext context)
	{
	}

	private void onOkClick(GameObject go)
	{
		this.CloseMe();
	}

	private void OnBackClick()
	{
		this.CloseMe();
	}

}

File: uLab/Assets/Lua/UI/DialogWindow.lua

local gameObject

DialogWindow = {}

function DialogWindow.OnInit(obj)
	gameObject = obj
	local ok = Util.FindGameObject(gameObject, 'buttons/button1')
	UIEventListener.SetOnClick(ok, onOkClick)
	local cancel = Util.FindGameObject(gameObject, 'buttons/back')
	UIEventListener.SetOnClick(cancel, onCancelClick)
end

function DialogWindow.OnEnter()
	
end

function DialogWindow.OnExit()
	
end

function DialogWindow.OnPause()
	
end

function DialogWindow.OnResume()
	
end

function onOkClick(go)
	CloseWindow(GameUI.dialog)
end

function onCancelClick(go)
	CloseWindow(GameUI.dialog)
end
